components/basic/Input.tsx

```diff
@@ -6,11 +6,12 @@ import { createSlug } from '@root/common/utilities';
 import styles from './Input.module.scss';
 
 export default function Input(props: {
+  id?: string,
   label?: any,
   inputClassName?: string,
   labelClassName?: string,
 } & React.ComponentPropsWithRef<'input'>) {
-  let id = (props.id ? props.id + '-' : '') + createSlug(props.label);
+  const id = props.id || 'input-' + Math.floor(Math.random() * 100000);
```

Comment:
hmm, do you know if this is a good practice? AFAIK the ID needs to be consistent to prevent unnecessary re-renders, but in this case, wouldn't it change every time the component rerenders as it re-computes `Math.random()`?

Reply:
this issue doesn't come up since the page is `'use client'`, however it definitely is kinda strange. the intention was to avoid requiring an id to be specified every time the input components are used, while still allowing the label to be associated with the inner input element. but maybe requiring an id just makes sense. thoughts?

Reply:
yeah I'd recommend not using `random` here, just allow the parent component to pass in the `id`. It can be a string type so can just make it the field label or whatever. https://stackoverflow.com/questions/29808636/when-giving-unique-keys-to-components-is-it-okay-to-use-math-random-for-gener

Reply:
i'll make an issue for this to be cleaned up later